src/frontend/hooks/hasProgress.ts

```diff
@@ -51,5 +61,23 @@ export const hasProgress = (appName: string) => {
     }
   }, [appName])
 
-  return [progress, previousProgress]
+  useEffect(() => {
+    if (isExtracting) {
+      setProgress(nullProgress) // reset progress to 0
+    }
+  }, [])
```

Comment:
I know this is legacy code but we don't need this useEffect. We can derive the value:

```tsx
return {
    progress: isExtracting ? nullProgress : progress,
    previousProgress,
    etaInMs: convertEtaToMs(progress.eta || '00:00:00')
}
```

Reply:
context: https://react.dev/learn/you-might-not-need-an-effect#updating-state-based-on-props-or-state

Reply:
Cool, will check this out

Reply:
This makes the progress always 0 when extracting, so I think we need the use effect because then it resets only once.

Reply:
Oh, gotcha, we need to reset it once. I think that's fine, then, either way, it seems this pattern is the recommended https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes, but tbh I've never used it.